### PR TITLE
fix: Display 2FA on Login Modal

### PR DESCRIPTION
**Pull request by victorgcramos**
This diff fixes a bug introduced by #2638, where modal overlays weren't
triggering visual updates, hence, modals were not being replaced, and the
2FA digits input couldn't be seen:

The error can be reproduced when logging in with a 2FA active account
using the "What are your thoughts" login button.

2fa-login-modal

- Also, adds e2e tests to cover this case.

Looks like this now:
2fa-login-modal-after

### CODE REVIEW

**Review by lukebp — APPROVED** [review]
tACK

Needs approval from @tiagoalvesdulce.

File: src/components/ModalProvider.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const ModalProvider = ({ children }) => {
   const modalToDisplay = useRef(initialState);
   // Modals are refs, and refs doesn't trigger component re-render. This
   // will force the state update, hence a modal update.
-  const [, setIsDisplaying] = useState(false);
+  const [, toggleDisplay] = useState(Date.now());
```
Comment (tiagoalvesdulce) [inline src/components/ModalProvider.jsx:18]: why are you using `Date.now()`?

**Review by tiagoalvesdulce — COMMENTED** [review]
Left an inline comment

File: src/components/ModalProvider.jsx (hunk 1 above)
Comment (victorgcramos) [reply src/components/ModalProvider.jsx:18]: It's a method that I am sure that will return me different values for any call. The purpose here is to force a component re-render. Do you have a better idea?

Comment (tiagoalvesdulce) [reply src/components/ModalProvider.jsx:18]: Yeah, you can use a dispatch function update. https://stackoverflow.com/questions/46240647/react-how-to-force-a-function-component-to-render/53837442#53837442

Comment (victorgcramos) [reply src/components/ModalProvider.jsx:18]: nice! thanks!

**Review by tiagoalvesdulce — APPROVED** [review]
tACK

**Pull request merged**